test/sqllogictest/persist-fast-path.slt

```diff
@@ -0,0 +1,98 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+mode cockroach
+
+simple conn=mz_system,user=mz_system
+ALTER SYSTEM SET persist_fast_path_limit = 100
+----
+COMPLETE 0
+
+# Verify that the persist fast path only kicks in when it's expected to do so.
+
+statement ok
+CREATE TABLE events (
+    content text,
+    insert_ms numeric,
+    delete_ms numeric
+);
+
+# Applies when the limit is below some threshold. Mapping and
+# projecting is fine.
+
+query T multiline
+EXPLAIN SELECT * from events limit 10;
```

Comment:
A non-`EXPLAIN` query is needed here that confirms that correct data is actually being returned for eligible queries.